akka-remote/src/test/scala/akka/remote/artery/BindCanonicalAddressSpec.scala

```diff
@@ -46,7 +48,10 @@ trait BindCanonicalAddressBehaviors {
       implicit val sys = ActorSystem("sys", config.withFallback(commonConfig))
 
       getExternal should ===(address.toAkkaAddress("akka"))
-      getInternal should not contain (address.toAkkaAddress("akka"))
+      // May have selected the same random port
+      if (!getInternal.headOption.exists(_.port.exists(randomPort ⇒ randomPort == address.getPort))) {
+        getInternal should not contain address.toAkkaAddress("akka")
+      }
```

Comment:
Wouldn't this also make the test pass if the functionality was broken? The assortment that if bind-port isn't the same as this random port then check that they aren't the same seems fishy.

Reply:
Yeah you're right this is crappy. Not sure how to make this test reliable

Reply:
Maybe if it retried the bind in this case, then we would catch if it got broken but in the event it randomly chose the same port that'd pass?

Reply:
Yeah, that decreases the odds, will do it